celery/backends/redis.py

```diff
@@ -103,17 +103,22 @@ def _reconnect_pubsub(self):
         self.backend.client.connection_pool.reset()
         # task state might have changed when the connection was down so we
         # retrieve meta for all subscribed tasks before going into pubsub mode
-        metas = self.backend.client.mget(self.subscribed_to)
-        metas = [meta for meta in metas if meta]
-        for meta in metas:
-            self.on_state_change(self._decode_result(meta), None)
+        if self.subscribed_to:
+            metas = self.backend.client.mget(self.subscribed_to)
+            metas = [meta for meta in metas if meta]
+            for meta in metas:
+                self.on_state_change(self._decode_result(meta), None)
         self._pubsub = self.backend.client.pubsub(
             ignore_subscribe_messages=True,
         )
+        # subscribed_to maybe empty after on_state_change
         if self.subscribed_to:
             self._pubsub.subscribe(*self.subscribed_to)
         else:
-            self._pubsub.ping()
```

Comment:
what are the limitations of self._pubsub.ping() ?

Reply:
pubsub will receive an empty reply b'$0\r\n\r\n\0x00...', this has no effect on celery's functionality, but causes extra connections and memory usage.  The reason is not to send a ping, but when redis using hiredis to receive data, each connection has a 64 bytes bytearray. Without sending ping, there are no these empty replies. </br>
 This problem was found when analyzing the high memory after running high-frequency tasks for a long time----there are many bytearray(b'$0\r\n\r\n\0x00...') takes up some memory.  
The unexpected growth in memory is related to redis-py, not celery.
redis-py, I also made some changes [redis-py 3.5.3 less memory usage](https://github.com/uuip/redis-py/tree/3.5)

Reply:
OK can you add additional unit tests for the changes?

Reply:
A unit test that verifies this behavior would be nice to have.